plugins/janus_streaming.c

```diff
@@ -5590,6 +5590,20 @@ static int janus_streaming_rtsp_connect_to_server(janus_streaming_mountpoint *mp
 			JANUS_LOG(LOG_VERB, "  -- SSRC (video): %"SCNu32"\n", ssrc);
 			source->video_ssrc = ssrc;
 		}
+		if(strcmp(vhost, "0.0.0.0") == 0) {
+			/* SDP does not contain a valid host address, so check for an RTSP 'source' address instead */
+			const char *source = strstr(curldata->buffer, ";source=");
+			if(source != NULL) {
+				const char *start = source+strlen(";source=");
+				const char *semicolon = strchr(start, ';');
+				const size_t source_len = semicolon ? (size_t)(semicolon-start) : strlen(start);
+				if (source_len < sizeof(vhost)) {
```

Comment:
If there's a check for that, there should be a warning if the source exceeds that value: right now it's just ignored.